src/Microsoft.VisualStudio.Composition/Reflection/ResolverExtensions.cs

```diff
@@ -162,19 +162,15 @@ internal static void GetInputAssemblies(this TypeRef typeRef, ISet<AssemblyName>
                     GetInputAssemblies(typeArg, assemblies);
                 }
 
+                assemblies.Add(typeRef.AssemblyName);
+
                 // Base types may define [InheritedExport] attributes or otherwise influence MEF
                 // so we should include them as input assemblies.
                 // Resolving a TypeRef is a necessary cost in order to identify the transitive closure of base types.
-                var type = typeRef.Resolve();
-                foreach (var baseType in type.EnumTypeAndBaseTypes())
-                {
-                    assemblies.Add(baseType.GetTypeInfo().Assembly.GetName());
-                }
-
-                // Interfaces may also define [InheritedExport] attributes, metadata view filters, etc.
-                foreach (var iface in type.GetTypeInfo().GetInterfaces())
+                //var type = typeRef.Resolve();
+                foreach (var baseType in typeRef.BaseTypes)
```

Comment:
>BaseTypes [](start = 49, length = 9)

This includes interfaces, which we weren't including before. That puts our cache in a more likely position to be considered stale when assemblies change that only contribute interfaces. Can we filter out the interfaces in this enumeration?

Reply:
The previous code enumerated both in separate loops. The end effect here should be the same.